### PR TITLE
fix raft compile error in milvus and optize bitset filter

### PR DESCRIPTION
**Pull request by Presburger**
issue: #852

### CODE REVIEW

File: src/index/ivf_raft/ivf_raft.cuh [hunk 1]
```diff
@@ -123,10 +127,21 @@ write_back(const int32_t* pre_ids_block, int64_t* ids_before, float* dis_before,
     } else {
         ids_after[ty * k1 + tx] = -1;
         dis_after[ty * k1 + tx] = 1.0f / 0.0f;
-        ids_before[0] = -1;  // destory old ans;
+        ids_before[0] = -2;  // destory old ans;
```
Comment (foxspy) [inline src/index/ivf_raft/ivf_raft.cuh:130]: Use a single variable could be better.

File: src/index/ivf_raft/ivf_raft_config.h [hunk 2]
```diff
@@ -64,6 +70,12 @@ class RaftIvfPqConfig : public IvfPqConfig {
     CFG_STRING internal_distance_dtype;
     CFG_FLOAT preferred_shmem_carveout;
     KNOHWERE_DECLARE_CONFIG(RaftIvfPqConfig) {
+        KNOWHERE_CONFIG_DECLARE_FIELD(k)
+            .set_default(10)
+            .description("search for top k similar vector.")
+            .set_range(1, 1024)
+            .for_search();
+
```
Comment (foxspy) [inline src/index/ivf_raft/ivf_raft_config.h:78]: The upper bound value 1024 is limited by hardware. However, this limitation can be easily broken by optimizing software, which can reduce the explanation cost.

Comment (Presburger) [reply src/index/ivf_raft/ivf_raft_config.h:78]: Good Idead.

**Review by foxspy — APPROVED** [review]
/lgtm

**Pull request merged**